src/objects.c

```diff
@@ -2510,20 +2518,60 @@ static CK_RV prep_ec_find(P11PROV_CTX *ctx, const OSSL_PARAM params[],
     }
     findctx->numattrs = 0;
 
-    rv = param_to_attr(ctx, params, OSSL_PKEY_PARAM_PUB_KEY, &findctx->attrs[0],
-                       CKA_P11PROV_PUB_KEY, false);
-    if (rv != CKR_OK) {
-        return rv;
-    }
-    findctx->numattrs++;
-
     group = EC_GROUP_new_from_params(params, p11prov_ctx_get_libctx(ctx), NULL);
     if (!group) {
         P11PROV_raise(ctx, CKR_KEY_INDIGESTIBLE, "Unable to decode ec group");
         rv = CKR_KEY_INDIGESTIBLE;
         goto done;
     }
 
+    p = OSSL_PARAM_locate_const(params, OSSL_PKEY_PARAM_PUB_KEY);
+    if (!p) {
+        P11PROV_raise(ctx, CKR_KEY_INDIGESTIBLE, "Missing %s",
+                      OSSL_PKEY_PARAM_PUB_KEY);
+        EC_GROUP_free(group);
+        return CKR_KEY_INDIGESTIBLE;
+    }
+
+    if (((char *)p->data)[0] == '\x02') {
+        P11PROV_debug("OpenSSL 3.0.7 BUG - received compressed EC public key");
+        pub_key[0].key = OSSL_PKEY_PARAM_PUB_KEY;
+        pub_key[0].data_type = p->data_type;
+        pub_key[0].data = OPENSSL_malloc(PUB_KEY_SIZE);
```

Comment:
Instead of using  fixed value, just call EC_POINT_point2oct() with a NULL buffer, and it will give you the needed lenght, so then we can malloc just the necessary data.

Alternatively if 200 bytes is always enough space, just allocate the space on the stack via something like `unit8_t pub_data[MAX_EC_PUB_KEY_SIZE];`